src/modules/robot/arm_solutions/RotatableDeltaSolution.cpp

```diff
@@ -0,0 +1,305 @@
+	if (xyz_full_steps_per_rotation < 1) xyz_full_steps_per_rotation = 200.0; //try to prevent stupid config entries
+
+	xyz_microsteps = config->value((unsigned int)xyz_microsteps_checksum)->by_default(16.0f)->as_number();
+	if (xyz_microsteps < 1) xyz_microsteps = 16.0; //try to prevent stupid config entries
+
+	small_pulley_teeth = config->value((unsigned int)small_pulley_teeth_checksum)->by_default(16.0f)->as_number();
+	if (small_pulley_teeth < 1) small_pulley_teeth = 16.0; //try to prevent stupid config entries
+
+	big_pulley_teeth = config->value((unsigned int)big_pulley_teeth_checksum)->by_default(150.0f)->as_number();
+	if (big_pulley_teeth < 1) big_pulley_teeth = 150.0; //try to prevent stupid config entries
+
+	//
+	//We need to override the alpha_steps_per_mm, beta_steps_per_mm & gamma_steps_per_mm read from the config file
+	//as these must be calculated based on other parameters in the config file on the SD card.
+	//The formula is:
+	//     (xyz_full_steps_per_rotation*xyz_microsteps*(big_pulley_teeth/small_pulley_teeth))*(two_pi*delta_rf)
+	//
+	//which means:
+	//
+	//xyz_full_steps_per_rotation*xyz_microsteps gives the number of steps the stepper motor needs to rotate 360 degrees
+	//
+	//(big_pulley_teeth/small_pulley_teeth) gives the ratio of the large pulley to the small one so that we can determine
+	//the  number of rotations the stepper motor needs to rotate the large pulley fully 360 degrees
+	//
+	//and the two_pi*delta_rf gives the circumference of the large pulley itself.
+	//
+	//So the formula informs us the number of steps the stepper motor needs to move a mm on the large pulley.
+	//
+	//For example, a 0.9 degree stepper has 400 steps per revolution, driven using a 1/32 microstep DRV8825 driver chip
+	//with a 16 tooth small pulley on the stepper motor and a 150 tooth (equivalent) large pulley of radius 90mm results in:
+	//
+	//(400*32*(150/16))/(2 * pi * 90) = 212.206590789194 microsteps are required to move the effector attached to
+	//the large pulley 1mm in a linear sense.
+	//
+	//Lastly, if you are not using a small and large pulley system like the FirePick is, simply make these two values
+	//the same, BUT NOT ZERO.
+
+	init();
+}
+
+// inverse kinematics
+// helper functions, calculates angle theta1 (for YZ-pane)
+int RotatableDeltaSolution::delta_calcAngleYZ(float x0, float y0, float z0, float &theta)
+{
+    float y1 = -0.5 * tan30 * delta_f; // f/2 * tan 30
+    y0 -= 0.5 * tan30       * delta_e; // shift center to edge
+    // z = a + b*y
+    float a = (x0*x0 + y0*y0 + z0*z0 + delta_rf*delta_rf - delta_re*delta_re - y1*y1) / (2.0*z0);
+    float b = (y1-y0)/z0;
+
+    float d = -(a+b*y1)*(a+b*y1) + delta_rf*(b*b*delta_rf+delta_rf); // discriminant
+    if (d < 0) return -1;                                            // non-existing point
+
+    float yj = (y1 - a*b - sqrt(d))/(b*b + 1.0);                     // choosing outer point
```

Comment:
OK, looks like I copied the wrong file.

I will do a clean compile after your recommended changes, load the firmware onto the FPD and check that everything works before I update git.

Cheers
Douglas
> On 10/06/2015, at 2:49 pm, Jim Morris <notifications@github.com <mailto:notifications@github.com>> wrote:
> 
> delta_calcAngleYZ